### PR TITLE
Feature/package refactor

### PR DESCRIPTION
**Pull request by sambles**
## Changes 
* Added backwards compatibilty so older imports work from moved packages / files 
 `from oasislmf.model_execution.bash import genbash` will import from `from oasislmf.execution.bash import genbash`

* created `computation` dir which holds all of the logic from `cli` and `OasisManager`
```
computation/
├── base.py
├── data
│   ├── gen_test_data
│   │   ├── gen_data_and_run_ktools.sh
│   │   └── generate_test_data.py
│   └── rtree.py
├── generate
│   ├── files.py
│   ├── keys.py
│   └── losses.py
├── helper
│   ├── autocomplete.py
│   ├── cookiecutter.py
├── hooks
│   ├── post_analysis.py
│   └── pre_analysis.py
└── run
    ├── exposure.py
    ├── model.py
    └── platform.py
```

* Moved the following: 
```
All root docs -> ./docs 
root scripts -> ./scripts

oasislmf/model_execution  --> oasislmf/execution
oasislmf/model_preparation --> oasislmf/preparation
oasislmf/api -> oasislmf/platform
oasislmf/model_testing -->  oasislmf/validation
oasislmf/gen_test_data -->  oasislmf/computation/data/gen_test_data
```

* split out the lookup code from `model_preperration/lookup.py` to:
```
oasislmf/lookup/
├── base.py
├── factory.py
├── interface.py
└── rtree.py
```
* Added "--keys-errors-csv <file>" to generate-oasis-files

* Added function signatures to OasisManager interface functions 
```
In [1]: from oasislmf.manager import OasisManager 
In [2]: OasisManager.generate_losses?                                                                                                                                                                                                                                                                                          
Signature:
OasisManager.generate_losses(
    oasis_files_dir=None,
    analysis_settings_json=None,
    user_data_dir=None,
    model_data_dir=None,
    model_run_dir=None,
    model_package_dir=None,
    ktools_alloc_rule_gul=None,
    ktools_disable_guard=None,
    ktools_fifo_relative=None,
    ktools_debug=None,
    ktools_legacy_stream=None,
    model_custom_gulcalc=None,
    ktools_num_processes=-1,
    ktools_alloc_rule_il=2,
    ktools_alloc_rule_ri=3,
)
Docstring:
Generates losses using the installed ktools framework given Oasis files,
model analysis settings JSON file, model data and model package data.
  ... etc
```

* Updated the Platform API command line calls 

## To do 
* Make Rtree an optional dependency and import

### CODE REVIEW

File: oasislmf/computation/run/model.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        get_analysis_settings(self.analysis_settings_json)
+        if self.model_settings_json:
+            get_model_settings(self.model_settings_json)
+
+        # Check input expsoure
+        required_ri_paths = [self.oed_info_csv , self.oed_scope_csv]
+        il = True if self.oed_accounts_csv else False
+        ri = all(required_ri_paths) and il
+        if any(required_ri_paths) and not ri:
+            raise OasisException(
```
Comment (sstruzik) [inline oasislmf/computation/run/model.py:68]: I think you haven't imported OasisException here

File: oasislmf/computation/run/model.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+)
+
+from ...utils.path import empty_dir
+
+class RunModel(ComputationStep):
+    """
+    Run models end to end.
+    """
+
+    chained_commands = [
```
Comment (sstruzik) [inline oasislmf/computation/run/model.py:26]: I prefer the name that you use for this feature. I think it is a better one that the one I used computation_steps.
if you look at what is in Computation Step, what I had in mind was
step_params was a variable that indicate the specific params of a step
computation_steps was the list of sub steps
and get_params was returning the full recursive list of all params.

File: oasislmf/computation/run/model.py (hunk 1 above)
Comment (sambles) [reply oasislmf/computation/run/model.py:66]: Thanks for spotting, fixed

File: oasislmf/computation/run/model.py (hunk 2 above)
Comment (sambles) [reply oasislmf/computation/run/model.py:34]: Ah okey I see!  I've switched the name to `chained_commands` and its now using the `all_params()` call from the base class

**Review by sstruzik — APPROVED** [review]
Looks good, impressive revamp :)

**Pull request merged**